src/Microsoft.Extensions.Http.Polly/DependencyInjection/PollyHttpClientBuilderExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using Microsoft.Extensions.Http;
+using Polly;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class PollyHttpClientBuilderExtensions
+    {
+        public static IHttpClientBuilder AddPolicyHandler(this IHttpClientBuilder builder, IAsyncPolicy policy)
+        {
+            //builder.AddHttpMessageHandler(() => new PolicyHttpMessageHandler(createPolicy()));
+            return builder;
+        }
+
+        public static IHttpClientBuilder AddPolicyHandler(this IHttpClientBuilder builder, IAsyncPolicy<HttpResponseMessage> policy)
+        {
+            //builder.AddHttpMessageHandler(() => new PolicyHttpMessageHandler(createPolicy()));
+            return builder;
+        }
+
+        public static IHttpClientBuilder AddExceptionPolicyHandler(this IHttpClientBuilder builder, Func<PolicyBuilder, IAsyncPolicy> createPolicy)
+        {
+            var policyBuilder = Policy.Handle<HttpRequestException>();
+            var policy = createPolicy(policyBuilder);
+
+            //builder.AddHttpMessageHandler(() => new PolicyHttpMessageHandler(createPolicy()));
+            return builder;
+        }
+
+        public static IHttpClientBuilder AddExceptionPolicyHandler(this IHttpClientBuilder builder, Func<PolicyBuilder, IAsyncPolicy<HttpResponseMessage>> createPolicy)
```

Comment:
Point of detail: this overload is not needed.  Polly fluent syntax happens to dictate that `PolicyBuilder` must already be generic `PolicyBuilder<TResult>` if the eventual policy is to be generic in `TResult`. (no configuration overloads flow from `PolicyBuilder` to `IAsyncPolicy<TResult>`.)

Reply:
Great! will remove this in the next iteration.